miio/device.py

```diff
@@ -136,6 +140,12 @@ def info(self, *, skip_cache=False) -> DeviceInfo:
 
         return self._fetch_info()
 
+    def initialize(self) -> None:
+        """This method implicitly populates the internal data structures needed
+        prior accessing `info()`, `sensors()`, `settings()`, or `actions()`."""
+        self.info()
```

Comment:
Alright, I removed the initialize method, and moved it back to the info() call.
I changed the other parts of the library to call _fetch_info() instead of info() to prevent these circular errors.